# **ETL – Lifestyle & Wellbeing Dataset**

## Objectives

* Objectives
- Load the Lifestyle & Wellbeing dataset from the local `dataset/raw` folder.
- Clean and transform the data (types, missing values, basic feature renaming).
- Save a cleaned version into `dataset/clean` for later analysis.

## Inputs
- `../dataset/raw/Wellbeing_and_lifestyle_data_Kaggle.csv`

## Outputs
- `../dataset/clean/lifestyle_wellbeing_clean.csv`

## Additional Comments

* TBD (currently none)



---

## 1. Set project directory

To avoid path errors, I first set the current working directory to the project root.  
All file paths used later (e.g. `dataset/raw` and `dataset/clean`) are defined **relative to this folder**.

In [61]:
import os
from pathlib import Path

project_root = Path("/Users/giaaxa/data analytics project /lifestyle_wellbeing_analysis")
os.chdir(project_root)

print("CWD now:", Path.cwd())
print("Contents here:", [p.name for p in Path.cwd().iterdir()])

CWD now: /Users/giaaxa/data analytics project /lifestyle_wellbeing_analysis
Contents here: ['setup.sh', 'requirements.txt', '.slugignore', 'dataset', 'README.md', 'jupyter_notebooks', '.gitignore', '.venv', '.python-version', 'Procfile', '.git']


## 2. Load the raw Lifestyle & Wellbeing dataset

Here I load the original CSV file from the `dataset/raw` folder into a pandas DataFrame.
This DataFrame (`df_raw`) is kept unchanged so I can always refer back to the original data.

In [63]:
import pandas as pd
import numpy as np

raw_path = "dataset/raw/Wellbeing_and_lifestyle_data_Kaggle.csv"

df_raw = pd.read_csv(raw_path)
df_raw.head()


Timestamp  FRUITS_VEGGIES DAILY_STRESS  PLACES_VISITED  CORE_CIRCLE  \
0    7/7/15               3            2               2            5   
1    7/7/15               2            3               4            3   
2    7/7/15               2            3               3            4   
3    7/7/15               3            3              10            3   
4    7/7/15               5            1               3            3   

   SUPPORTING_OTHERS  SOCIAL_NETWORK  ACHIEVEMENT  DONATION  BMI_RANGE  ...  \
0                  0               5            2         0          1  ...   
1                  8              10            5         2          2  ...   
2                  4              10            3         2          2  ...   
3                 10               7            2         5          2  ...   
4                 10               4            2         4          2  ...   

   SLEEP_HOURS  LOST_VACATION  DAILY_SHOUTING  SUFFICIENT_INCOME  \
0            7              5               5                  1   
1            8              2               2                  2   
2            8             10               2                  2   
3            5              7               5                  1   
4            7              0               0                  2   

   PERSONAL_AWARDS  TIME_FOR_PASSION  WEEKLY_MEDITATION         AGE  GENDER  \
0                4                 0                  5    36 to 50  Female   
1                3                 2                  6    36 to 50  Female   
2                4                 8                  3    36 to 50  Female   
3                5                 2                  0  51 or more  Female   
4                8                 1                  5  51 or more  Female   

   WORK_LIFE_BALANCE_SCORE  
0                    609.5  
1                    655.6  
2                    631.6  
3                    622.7  
4                    663.9  

[5 rows x 24 columns]

## 3. Inspect raw data structure and column names

In this step I check:
- Number of rows and columns
- Data types for each column
- Basic column names

This gives a baseline view before cleaning and transformation.

In [64]:
df_raw.info()
df_raw.columns


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15972 entries, 0 to 15971
Data columns (total 24 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Timestamp                15972 non-null  object 
 1   FRUITS_VEGGIES           15972 non-null  int64  
 2   DAILY_STRESS             15972 non-null  object 
 3   PLACES_VISITED           15972 non-null  int64  
 4   CORE_CIRCLE              15972 non-null  int64  
 5   SUPPORTING_OTHERS        15972 non-null  int64  
 6   SOCIAL_NETWORK           15972 non-null  int64  
 7   ACHIEVEMENT              15972 non-null  int64  
 8   DONATION                 15972 non-null  int64  
 9   BMI_RANGE                15972 non-null  int64  
 10  TODO_COMPLETED           15972 non-null  int64  
 11  FLOW                     15972 non-null  int64  
 12  DAILY_STEPS              15972 non-null  int64  
 13  LIVE_VISION              15972 non-null  int64  
 14  SLEEP_HOURS           

Index(['Timestamp', 'FRUITS_VEGGIES', 'DAILY_STRESS', 'PLACES_VISITED',
       'CORE_CIRCLE', 'SUPPORTING_OTHERS', 'SOCIAL_NETWORK', 'ACHIEVEMENT',
       'DONATION', 'BMI_RANGE', 'TODO_COMPLETED', 'FLOW', 'DAILY_STEPS',
       'LIVE_VISION', 'SLEEP_HOURS', 'LOST_VACATION', 'DAILY_SHOUTING',
       'SUFFICIENT_INCOME', 'PERSONAL_AWARDS', 'TIME_FOR_PASSION',
       'WEEKLY_MEDITATION', 'AGE', 'GENDER', 'WORK_LIFE_BALANCE_SCORE'],
      dtype='object')

## 4. Create working copy and rename columns to `snake_case`

I create a working copy called `df` and rename all variables to `snake_case`.
This makes the dataset easier to work with programmatically and keeps naming consistent.

In [65]:
df = df_raw.copy()

# Renaming to snake_case, easier to work with later
df = df.rename(columns={
    "Timestamp": "timestamp",
    "FRUITS_VEGGIES": "fruits_veggies",
    "DAILY_STRESS": "daily_stress",
    "PLACES_VISITED": "places_visited",
    "CORE_CIRCLE": "core_circle",
    "SUPPORTING_OTHERS": "supporting_others",
    "SOCIAL_NETWORK": "social_network",
    "ACHIEVEMENT": "achievement",
    "DONATION": "donation",
    "BMI_RANGE": "bmi_range",
    "TODO_COMPLETED": "todo_completed",
    "FLOW": "flow",
    "DAILY_STEPS": "daily_steps",
    "LIVE_VISION": "live_vision",
    "SLEEP_HOURS": "sleep_hours",
    "LOST_VACATION": "lost_vacation",
    "DAILY_SHOUTING": "daily_shouting",
    "SUFFICIENT_INCOME": "sufficient_income",
    "PERSONAL_AWARDS": "personal_awards",
    "TIME_FOR_PASSION": "time_for_passion",
    "WEEKLY_MEDITATION": "weekly_meditation",
    "AGE": "age",
    "GENDER": "gender",
    "WORK_LIFE_BALANCE_SCORE": "work_life_balance_score"
})

df.head()


timestamp  fruits_veggies daily_stress  places_visited  core_circle  \
0    7/7/15               3            2               2            5   
1    7/7/15               2            3               4            3   
2    7/7/15               2            3               3            4   
3    7/7/15               3            3              10            3   
4    7/7/15               5            1               3            3   

   supporting_others  social_network  achievement  donation  bmi_range  ...  \
0                  0               5            2         0          1  ...   
1                  8              10            5         2          2  ...   
2                  4              10            3         2          2  ...   
3                 10               7            2         5          2  ...   
4                 10               4            2         4          2  ...   

   sleep_hours  lost_vacation  daily_shouting  sufficient_income  \
0            7              5               5                  1   
1            8              2               2                  2   
2            8             10               2                  2   
3            5              7               5                  1   
4            7              0               0                  2   

   personal_awards  time_for_passion  weekly_meditation         age  gender  \
0                4                 0                  5    36 to 50  Female   
1                3                 2                  6    36 to 50  Female   
2                4                 8                  3    36 to 50  Female   
3                5                 2                  0  51 or more  Female   
4                8                 1                  5  51 or more  Female   

   work_life_balance_score  
0                    609.5  
1                    655.6  
2                    631.6  
3                    622.7  
4                    663.9  

[5 rows x 24 columns]

## 5. Convert data types (dates and numeric variables)

In this step I:
- Convert `timestamp` to a proper datetime type.
- Ensure all score and count variables are numeric.
- Coerce any invalid entries to `NaN` so they can be handled explicitly later.

In [66]:
# Converting timestamp
df["timestamp"] = pd.to_datetime(df["timestamp"])

numeric_cols = [
    "fruits_veggies",
    "daily_stress",
    "places_visited",
    "core_circle",
    "supporting_others",
    "social_network",
    "achievement",
    "donation",
    "bmi_range",
    "todo_completed",
    "flow",
    "daily_steps",
    "live_vision",
    "sleep_hours",
    "lost_vacation",
    "daily_shouting",
    "sufficient_income",
    "personal_awards",
    "time_for_passion",
    "weekly_meditation",
    "age",
    "work_life_balance_score"
]

df[numeric_cols] = df[numeric_cols].apply(pd.to_numeric, errors="coerce")
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15972 entries, 0 to 15971
Data columns (total 24 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   timestamp                15972 non-null  datetime64[ns]
 1   fruits_veggies           15972 non-null  int64         
 2   daily_stress             15971 non-null  float64       
 3   places_visited           15972 non-null  int64         
 4   core_circle              15972 non-null  int64         
 5   supporting_others        15972 non-null  int64         
 6   social_network           15972 non-null  int64         
 7   achievement              15972 non-null  int64         
 8   donation                 15972 non-null  int64         
 9   bmi_range                15972 non-null  int64         
 10  todo_completed           15972 non-null  int64         
 11  flow                     15972 non-null  int64         
 12  daily_steps              15972 n

/var/folders/z2/cdr6_mb16bz38cj75fm9fl4m0000gn/T/ipykernel_83881/24149206.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df["timestamp"] = pd.to_datetime(df["timestamp"])


## 6. Remove duplicate records

To avoid biasing the analysis, I remove rows that are exact duplicates across all columns.

- First I count how many full-row duplicates exist.
- Then I drop them and report the change in row count.

In [67]:
# Checking for duplicates
print("Duplicate rows:", df.duplicated().sum())
df = df.drop_duplicates()

print("Rows before:", len(df_raw))
print("Rows after dropping duplicates:", len(df))

Duplicate rows: 482
Rows before: 15972
Rows after dropping duplicates: 15490


I identified 482 fully duplicated rows in the raw dataset and removed them using drop_duplicates(), leaving 15,490 unique observations.

## 7. Save cleaned dataset to `dataset/clean`

Finally, I save the transformed and de-duplicated dataset as
`lifestyle_wellbeing_clean.csv` in the `dataset/clean` folder.

This file will be used by the analysis notebook for visualisation and modelling.

In [68]:
from pathlib import Path


clean_dir = Path("dataset") / "clean"
clean_dir.mkdir(parents=True, exist_ok=True)  

clean_path = clean_dir / "lifestyle_wellbeing_clean.csv"

df.to_csv(clean_path, index=False)

clean_path

PosixPath('dataset/clean/lifestyle_wellbeing_clean.csv')